src/Hl7.Fhir.ElementModel.Tests/TypedElementOnSourceNodeTests.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Linq;
+using Hl7.Fhir.Serialization;
+using Hl7.Fhir.Specification;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Hl7.Fhir.ElementModel.Tests
+{
+    [TestClass]
+    public class TypedElementOnSourceNodeTests
+    {
+        private readonly IStructureDefinitionSummaryProvider provider = new PocoStructureDefinitionSummaryProvider();
+
+        [TestMethod]
```

Comment:
Use here the the following attribute: `[ExpectedException(typeof(StructuralTypeException))]`
Then you do not have to use the try..except

Reply:
Changed as requested